### PR TITLE
br: fix S3 backup endpoint suffix /

### LINKED ISSUE #30128: During S3 backup, errors may occur if there is "/" at the end of the endpoint
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

when S3 backup or restore, If "/" is added last in the endpoint, an error may occur

2. What did you expect to see?

Whether "/" is added last, the backup can be successful

3. What did you see instead?

4. What version of BR and TiDB/TiKV/PD are you using?

br -4.0.11
br -5
The above problem occurs only when there is data in instance. Therefore, when br version is 4.0.11, this problem will not occur when backing up empty instances, because the system library will not be backed up; However, when the br version is 5, even for empty instances, Br the system library will be backed up during backup.

5. Operation logs
   - Please upload `br.log` for BR if possible
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `topology.yml` if deployed by TiUP

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus if possible

### PR DESCRIPTION
**Pull request by tongtongyin**
### What problem does this PR solve?

Issue Number: close #30128 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by kennytm** [comment]
thanks for the PR, however we've already got a fix at #30138 pending to be merged.

File: br/pkg/storage/s3.go [hunk 1]
```diff
@@ -189,6 +189,7 @@ func (options *S3BackendOptions) parseFromFlags(flags *pflag.FlagSet) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	options.Endpoint = strings.Trim(options.Endpoint, "/")
```
Comment (kennytm) [inline br/pkg/storage/s3.go:192]:
```suggestion
	options.Endpoint = strings.TrimSuffix(options.Endpoint, "/")
```

File: ddl/options_test.go [hunk 2]
```diff
@@ -15,22 +15,21 @@
 package ddl_test
 
 import (
+	"github.com/stretchr/testify/require"
```
Comment (kennytm) [inline ddl/options_test.go:18]:
```
[2021-11-30T08:07:27.397Z] ddl/options_test.go:18: File is not `goimports`-ed (goimports)
[2021-11-30T08:07:27.397Z] 	"github.com/stretchr/testify/require"
```

let's not do the test-infra migration in this PR.

**Review by fengou1 — COMMENTED** [review]
Here we try to get flag s3.endpoint, what if -s "s3://bank/shenma-without-stats?access-key=admin&secret-access-key=Passadmin&endpoint=http://minio.pingcap.net:9000/&force-path-style=true"

File: ddl/options_test.go (hunk 2 above)
Comment (tongtongyin) [reply ddl/options_test.go:18]: ok

**Comment by tongtongyin** [comment]
I resubmitted a PR and closed the current one

**Comment by tongtongyin** [comment]
#30530

**Pull request closed without merging**